test/v2/controllers/validators/RetrieveUkPropertyAnnualSubmissionValidatorFactorySpec.scala

```diff
@@ -0,0 +1,114 @@
+
+  private val parsedNino       = Nino(validNino)
+  private val parsedBusinessId = BusinessId(validBusinessId)
+  private val parsedTaxYear    = TaxYear.fromMtd(validTaxYear)
+
+  private val validatorFactory = new RetrieveUkPropertyAnnualSubmissionValidatorFactory(mockAppConfig)
+
+  private def validator(nino: String, businessId: String, taxYear: String) =
+    validatorFactory.validator(nino, businessId, taxYear)
+
+  class SetUp {
+
+    MockAppConfig.minimumTaxV2Uk
+      .returns(2021)
+      .anyNumberOfTimes()
+
+  }
+
+  "validator" should {
+    "return a request object" when {
+      "valid request data is supplied" in new SetUp {
+        val result: Either[ErrorWrapper, RetrieveUkPropertyAnnualSubmissionRequestData] =
+          validator(validNino, validBusinessId, validTaxYear).validateAndWrapResult()
+
+        result shouldBe Right(RetrieveUkPropertyAnnualSubmissionRequestData(parsedNino, parsedBusinessId, parsedTaxYear))
+      }
+    }
+
+    "return an ErrorWrapper" when {
+      "a single validation error occurs" in new SetUp {
+        val result: Either[ErrorWrapper, RetrieveUkPropertyAnnualSubmissionRequestData] =
+          validator("invalid nino", validBusinessId, validTaxYear).validateAndWrapResult()
+
+        result shouldBe Left(ErrorWrapper(correlationId, NinoFormatError, None))
+      }
+
+    }
+
+    "passed an incorrectly formatted taxYear" in new SetUp {
+      val result: Either[ErrorWrapper, RetrieveUkPropertyAnnualSubmissionRequestData] =
+        validator(validNino, validBusinessId, "202324").validateAndWrapResult()
+
+      result shouldBe Left(ErrorWrapper(correlationId, TaxYearFormatError))
+
+    }
+
+    "passed an incorrectly formatted businessId" in new SetUp {
+      val result: Either[ErrorWrapper, RetrieveUkPropertyAnnualSubmissionRequestData] =
+        validator(validNino, "invalid business id", validTaxYear).validateAndWrapResult()
+
+      result shouldBe Left(ErrorWrapper(correlationId, BusinessIdFormatError))
+    }
+
+    "passed an unsupported taxYear" in new SetUp {
+      val result: Either[ErrorWrapper, RetrieveUkPropertyAnnualSubmissionRequestData] =
+        validator(validNino, validBusinessId, "2019-20").validateAndWrapResult()
+
+      result shouldBe Left(ErrorWrapper(correlationId, RuleTaxYearNotSupportedError))
+    }
+
+    "passed a taxYear spanning an invalid tax year range" in new SetUp {
+      val result: Either[ErrorWrapper, RetrieveUkPropertyAnnualSubmissionRequestData] =
+        validator(validNino, validBusinessId, "2021-23").validateAndWrapResult()
+
+      result shouldBe Left(ErrorWrapper(correlationId, RuleTaxYearRangeInvalidError))
+    }
+
+    "return multiple validation errors" when {
+      "the request has multiple issues (path parameters)" in new SetUp {
+        val result: Either[ErrorWrapper, RetrieveUkPropertyAnnualSubmissionRequestData] =
+          validator("Nino", "BusinessId", "2021/22").validateAndWrapResult()
```

Comment:
for the sake of consistency with the other tests in the other PRs and for explicitness can we do:
```suggestion
          validator("invalid", "invalid", "invalid").validateAndWrapResult()
```